### DISCUSSION TITLE
How do I associate context when using Models.ChatGpt3_5Turbo?

### DISCUSSION THREAD

**Discussion by traitors09** [betalgo/openai#130] [Q&A] [answered]
How do I associate context when using Models.ChatGpt3_5Turbo?

`
var completionResult = await sdk.ChatCompletion.CreateCompletion(new ChatCompletionCreateRequest
{

    Messages = new List<ChatMessage>
    {
        ChatMessage.FromSystem("You are a helpful assistant."),
        ChatMessage.FromUser("Who won the world series in 2020?"),
        ChatMessage.FromAssistance("The Los Angeles Dodgers won the World Series in 2020."),
        ChatMessage.FromUser("Where was it played?")
    },

    Model = Models.ChatGpt3_5Turbo,
    MaxTokens = 50//optional

});
if (completionResult.Successful)
{
   Console.WriteLine(completionResult.Choices.First().Message.Content);

}
`

The CreateCompletion function seems to produce a new chat every time

**Comment by Frogley** [comment, marked as answer]
According to my understanding, what gpt-3.5-turbo provides is "chat completion" capability, rather than "conversation" capability. The List of ChatMessage is the context of the conversation, based on which gpt infers the subsequent response.

**Reply by traitors09** [reply]
Yes, you're right. The official documentation states that in order to synchronize context, you need to send the last chat each time or at least a partial summary of the last chat

**Reply by traitors09** [reply]
Document text：
**Including the conversation history helps when user instructions refer to prior messages. In the example above, the user’s final question of “Where was it played? ” only makes sense in the context of the prior messages about the World Series of 2020. Because the models have no memory of past requests, all relevant information must be supplied via the conversation. If a conversation cannot fit within the model’s token limit, it will need to be shortened in some way.**
